sklearn/metrics/_ranking.py

```diff
@@ -1419,3 +1419,100 @@ def ndcg_score(y_true, y_score, k=None, sample_weight=None, ignore_ties=False):
     _check_dcg_target_type(y_true)
     gain = _ndcg_sample_scores(y_true, y_score, k=k, ignore_ties=ignore_ties)
     return np.average(gain, weights=sample_weight)
+
+
+def top_k_accuracy_score(y_true, y_score, k=2, normalize=True):
+    """Top k Accuracy classification score.
+
+    This metric computes the number of times where the correct label is among
+    the top `k` labels predicted (ranked by predicted scores). Note that
+    multilabel classification case isn't covered here.
+
+    Parameters
+    ----------
+    y_true : array-like of shape (n_samples,)
+        True labels.
+
+    y_score : array-like of shape (n_samples, n_classes)
+        Target scores. These can be either probability estimates or
+        non-thresholded decision values as returned by :term:`predict_proba` or
+        :term:`decision_function`, respectively.
+
+    k : int, default=2
+        Number of guesses allowed to find the correct label.
+
+    normalize : bool, default=True
+        If `True`, return the fraction of correctly classified samples.
+        Otherwise, return the number of correctly classified samples.
+
+    Returns
+    -------
+    score : float
+        The top-k accuracy score. The best performance is 1 with
+        `normalize == True` and the number of samples with
+        `normalize == False`.
+
+    See also
+    --------
+    accuracy_score
+
+    Notes
+    -----
+    In cases where two or more labels are assigned equal predicted scores,
+    the labels with the smallest indices will be chosen first. Then, the result
+    might be incorrect if the correct label falls after the threshold because
+    of that.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.metrics import top_k_accuracy_score
+    >>> y_true = np.array([0, 1, 2, 2])
+    >>> y_score = np.array([[0.5, 0.2, 0.2],
+    ...                     [0.3, 0.4, 0.2],
+    ...                     [0.2, 0.4, 0.3],
+    ...                     [0.7, 0.2, 0.1]])
+    >>> top_k_accuracy_score(y_true, y_score, k=2)
+    0.75
+    >>> top_k_accuracy_score(y_true, y_score, k=2, normalize=False)
+    3
+    >>> top_k_accuracy_score([0]*4, y_score, k=2)
+    0.75
+
+    """
+    if k == 1:
+        raise ValueError(
+            "'k'=1 is equivalent to 'metrics.accuracy_score'. Please, use "
+            "that method instead."
+        )
+
+    check_consistent_length(y_true, y_score)
+    y_type = type_of_target(y_true)
+
+    if y_type not in {'binary', 'multiclass'}:
+        raise ValueError(
+            f"Target type must be 'binary' or 'multiclass', got '{y_type}' "
+            "instead."
+        )
+
+    y_true = column_or_1d(y_true)
+    y_score = check_array(y_score)
+
+    n_classes = y_score.shape[1]
+    if k == n_classes:
+        raise ValueError(
+            f"'k'='n_classes' ({n_classes}) results in a perfect score and is "
+            "therefore meaningless."
+        )
+
+    infered_n_classes = y_true.max() + 1
```

Comment:
Do we make the assumption that the labels are in `0,1...` or am I missing something?